### PR TITLE
fix: TNLT-6532 opta widget showing error on mobile

### PR DESCRIPTION
**Pull request by kamiranoff**
- https://nidigitalsolutions.jira.com/secure/RapidBoard.jspa?rapidView=1334&projectKey=TNLT&selectedIssue=TNLT-6532
- fix Opta widget showing error on android

**Before:**
Screenshot_20210129_130409_uk co thetimes debug

**After:**
Screenshot_20210129_130340_uk co thetimes debug

### CODE REVIEW

File: packages/interactive-wrapper/src/interactive-wrapper.android.js [hunk 1]
```diff
@@ -20,10 +20,12 @@ class InteractiveWrapper extends Component {
...
   }
 
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
     this.state = {
       height: 1,
+      hasError: false,
+      hasLoaded: false,
```
Comment (Geoff-Ford) [inline packages/interactive-wrapper/src/interactive-wrapper.android.js:28]: Not a biggie but we generally use `isLoading` in other parts of TCN and default it to `true` then set it to `false` when loading complete.

Comment (kamiranoff) [reply packages/interactive-wrapper/src/interactive-wrapper.android.js:28]: I'll change it. Will make the if statement easier to read as well

File: packages/interactive-wrapper/src/interactive-wrapper.android.js [hunk 2]
```diff
@@ -62,9 +71,13 @@ class InteractiveWrapper extends Component {
       config: { dev, environment, platform, version },
       id,
     } = this.props;
-    const { height } = this.state;
+    const { height, hasError, hasLoaded } = this.state;
     const uri = `${editorialLambdaProtocol}${editorialLambdaOrigin}/${editorialLambdaSlug}/${id}?dev=${dev}&env=${environment}&platform=${platform}&version=${version}`;
 
+    if (hasError || !hasLoaded) {
+      return null;
+    }
```
Comment (Geoff-Ford) [inline packages/interactive-wrapper/src/interactive-wrapper.android.js:79]: This can all go on a single line.

**Review by Geoff-Ford — APPROVED** [review]
LGTM. Two minor points.

File: packages/interactive-wrapper/src/interactive-wrapper.android.js (hunk 2 above)
Comment (kamiranoff) [reply packages/interactive-wrapper/src/interactive-wrapper.android.js:79]: `if (hasError || isLoading) return null;` Prettier is happy with this if I remove the curly brackets.

Comment (Geoff-Ford) [reply packages/interactive-wrapper/src/interactive-wrapper.android.js:79]: Yep, it's really a style thing but we have generally conformed to this if possible. If the conditional code is longer than 1 line then we keep it in a block.

File: packages/interactive-wrapper/src/interactive-wrapper.android.js [hunk 3]
```diff
@@ -32,11 +34,18 @@ class InteractiveWrapper extends Component {
...
 
   onLoadEnd() {
+    this.setState({ isLoading: false });
     if (this.webview) {
       this.webview.postMessage("thetimes.co.uk", "*");
     }
   }
 
+  handleHttpError = (event) => {
+    if (event?.nativeEvent?.statusCode > 300) {
```
Comment (kamiranoff) [inline packages/interactive-wrapper/src/interactive-wrapper.android.js:44]: Maybe that should be `400`. Not sure if we would ever get 300s?

**Review by c-j-j — APPROVED** [review]

**Review by aronshamash — APPROVED** [review]

**Pull request merged**